pkg/controllers/workapplier/availability_tracker_test.go

```diff
@@ -983,6 +995,125 @@ func TestTrackInMemberClusterObjAvailabilityByGVR(t *testing.T) {
 	}
 }
 
+func TestServiceExportAvailability(t *testing.T) {
+	availableValidSvcExport := svcExportTemplate.DeepCopy()
+	availableValidSvcExport.Status = fleetnetworkingv1alpha1.ServiceExportStatus{
+		Conditions: []metav1.Condition{
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportValid),
+				Status: metav1.ConditionTrue,
+				Reason: "ServiceIsValid",
+			},
+		},
+	}
+
+	unavailableInvalidSvcExport := svcExportTemplate.DeepCopy()
+	unavailableInvalidSvcExport.Status = fleetnetworkingv1alpha1.ServiceExportStatus{
+		Conditions: []metav1.Condition{
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportValid),
+				Status: metav1.ConditionFalse,
+				Reason: "ServiceNotFound",
+			},
+		},
+	}
+
+	availableNoConflictSvcExport := svcExportTemplate.DeepCopy()
+	availableNoConflictSvcExport.Annotations["networking.fleet.azure.com/weight"] = "1"
+	availableNoConflictSvcExport.Status = fleetnetworkingv1alpha1.ServiceExportStatus{
+		Conditions: []metav1.Condition{
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportValid),
+				Status: metav1.ConditionTrue,
+				Reason: "ServiceIsValid",
+			},
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportConflict),
+				Status: metav1.ConditionFalse,
+				Reason: "NoConflictFound",
+			},
+		},
+	}
+
+	unavailableHasConflictSvcExport := svcExportTemplate.DeepCopy()
+	unavailableHasConflictSvcExport.Annotations["networking.fleet.azure.com/weight"] = "1"
+	unavailableHasConflictSvcExport.Status = fleetnetworkingv1alpha1.ServiceExportStatus{
+		Conditions: []metav1.Condition{
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportValid),
+				Status: metav1.ConditionTrue,
+				Reason: "ServiceIsValid",
+			},
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportConflict),
+				Status: metav1.ConditionTrue,
+				Reason: "ConflictFound",
+			},
+		},
+	}
+
+	unavailableInvalidNoConflictSvcExport := svcExportTemplate.DeepCopy()
+	unavailableInvalidNoConflictSvcExport.Annotations["networking.fleet.azure.com/weight"] = "1"
+	unavailableInvalidNoConflictSvcExport.Status = fleetnetworkingv1alpha1.ServiceExportStatus{
+		Conditions: []metav1.Condition{
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportValid),
+				Status: metav1.ConditionFalse,
+				Reason: "ServiceIneligible",
+			},
+			{
+				Type:   string(fleetnetworkingv1alpha1.ServiceExportConflict),
+				Status: metav1.ConditionTrue,
+				Reason: "ConflictFound",
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                                         string
+		svcExport                                    *fleetnetworkingv1alpha1.ServiceExport
```

Comment:
could you please make the annotation & status as the test input?

before run, you can always copy the template and set these fields. It's easier to figure out what's your input per each test.